core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -2134,11 +2163,26 @@ class ParallelScan<FunctorType, Kokkos::RangePolicy<Traits...>, Kokkos::Cuda> {
   //----------------------------------------
 
   __device__ inline void operator()(void) const {
-    if (!m_final) {
-      initial();
+#ifdef KOKKOS_IMPL_DEBUG_CUDA_SERIAL_EXECUTION
+    if (m_run_serial) {
```

Comment:
The `m_run_serial` data member is pointless since you still have to put code behind the `KOKKOS_IMPL_DEBUG_CUDA_SERIAL_EXECUTION` macro.  You might as well use `Kokkos::Impl::CudaInternal::cuda_use_serial_execution()` directly.

Reply:
No it is not pointless. I can't call cuda_use_serial_execution on the device since it will in the end query a static variable only available on the host.

Reply:
`cuda_use_serial_execution` is not accessible on the device since it querys a static variable only accessible on the host. That is why I need `m_run_serial`.